app/models/user.rb

```diff
@@ -27,4 +27,24 @@ def get_students(mod_num)
         student.user_profile if student.user_profile.current_mod == mod_num
       end.compact
   end
+
+  def find_students(search_term)
+    case search_term.keys
+    when ["first_name"]
+      
+      UserProfile.select("first_name, last_name, current_cohort, user_id")
```

Comment:
Would love ideas on how to refactor this,!

Reply:
My suggestions/questions are below. Apologies if it's a little long. I am happy to chat in more detail if you have questions or want to pair.
1.  Would this be better as class method on the UserProfile model rather than the User? I may be overlooking it but seems you are not accessing anything from User.

2. Because we are doing a name search it's best not surround the search term with the wildcard (%) and just put the wildcard on the end. If the DB was had millions of records, looking for any name that contained two characters anywhere inside that attribute could be very taxing on the DB.

3. Case when you can avoid if you pass down the first and last names as separate arguments. If they dont exist in the request then you can pass down an empty string in its place. Or you can assign them as variables inside the method on the model, maybe like below?.
```ruby
   first_name = "" ? search_term["first_name"] : search_term["first_name"]
   last_name = "" ? search_term["last_name"] : search_term["last_name"]
```
4. If you use empty strings when an optional query parameter does not exist, it is in effect only searching by the single term that does. However, if you use this route then it's best to ensure that you never allow both fields to be empty. 

5. Last note, you may want to ask the FE how they want results ordered. Without an order on the query, it will typically return them in the order of the PK. May be better to handle sorting on our end than theirs.